core/imt/test/testTBBGlobalControl.cxx

```diff
@@ -0,0 +1,51 @@
+#include "TROOT.h"
+#include "ROOT/RTaskArena.hxx"
+#include "ROOT/TThreadExecutor.hxx"
+#include "ROOTUnitTestSupport.h"
+#include "gtest/gtest.h"
+#include "tbb/global_control.h"
+
+#ifdef R__USE_IMT
+
+const unsigned maxConcurrency = ROOT::Internal::LogicalCPUBandwithControl();
+
+TEST(TBBGlobalControl, RTaskArena)
+{
+   auto gTAInstance = ROOT::Internal::GetGlobalTaskArena(maxConcurrency);
+   tbb::global_control c(tbb::global_control::max_allowed_parallelism, 2u);
+   gTAInstance.reset();
+   ROOT_EXPECT_WARNING(gTAInstance = ROOT::Internal::GetGlobalTaskArena(maxConcurrency);,
+                       "RTaskArenaWrapper",
+                       "tbb::global_control is active, limiting the number of parallel workers"
+                       "from this task arena available for execution.");
+}
+
+TEST(TBBGlobalControl, TThreadExecutor)
+{
+   // ***See them pass***
+   ROOT::TThreadExecutor executor{maxConcurrency};
+   executor.Map([]() { return 1; }, 10); // ParallelFor
+   std::vector<double> vd{0., 1., 2.};
+   executor.Reduce(vd, std::plus<double>{}); // ParallelReduce double
+   std::vector<float> vf{0., 1., 2.};
+   executor.Reduce(vf, std::plus<float>{}); // ParallelReduce float
+
+   tbb::global_control c(tbb::global_control::max_allowed_parallelism, 2u);
```

Comment:
This won't trigger warnings if `maxConcurrency` is <= 2, right? So maybe to avoid false positives:

```suggestion
      if (maxConcurrency <= 1)
         return; // skip this test on systems with only 1 core
      tbb::global_control c(tbb::global_control::max_allowed_parallelism, 1u);
```

Reply:
Good catch, thanks!